client/object_get.go

```diff
@@ -269,17 +279,18 @@ func (x *ObjectReader) Read(p []byte) (int, error) {
 //
 // Return errors:
 //   - [ErrMissingSigner]
-func (c *Client) ObjectGetInit(ctx context.Context, containerID cid.ID, objectID oid.ID, prm PrmObjectGet) (*ObjectReader, error) {
+func (c *Client) ObjectGetInit(ctx context.Context, containerID cid.ID, objectID oid.ID, prm PrmObjectGet) (object.Object, *ObjectReader, error) {
```

Comment:
The same way like in `ObjectPutInit`. In _put_ we save header transparent, thus in _get_ we may take them transparent too. It also helps to avoid missing `readHeader` call